code/identity/src/main/java/com/adobe/marketing/mobile/identity/IdentityExtension.java

```diff
@@ -1884,11 +1898,23 @@ void processPrivacyChange(final Event event, final Map<String, Object> eventData
             savePersistently(); // clear datastore
             getApi().createSharedState(packageEventData(), event);
         } else if (StringUtils.isNullOrEmpty(mid)) {
-            // When changing privacy status from optedout, need to generate new Experience Cloud ID
-            // for the user
-            // Queue up a request to sync the new ID with the Identity Service
-            Event syncEvent = createForcedSyncEvent();
-            getApi().dispatch(syncEvent);
+            // Need to generate new Experience Cloud ID for the user
+            ConfigurationSharedStateIdentity configSharedState =
+                    new ConfigurationSharedStateIdentity();
+
+            SharedStateResult configState =
+                    getApi().getSharedState(
+                                    IdentityConstants.EventDataKeys.Configuration.MODULE_NAME,
+                                    event,
+                                    false,
+                                    SharedStateResolution.LAST_SET);
+            if (configState != null) {
+                configSharedState.getConfigurationProperties(configState.getValue());
+            }
+
+            if (handleSyncIdentifiers(event, configSharedState, true)) {
```

Comment:
for both processPrivacyChange and resetIdentities ios calls syncIdentifiers(event:event) -> with default forceSync false 

https://github.com/adobe/aepsdk-core-ios/blob/main/AEPIdentity/Sources/IdentityState.swift#L348

Reply:
If the ECID is null, then that will force sync, so removing the "forceSync" flag is okay.